package.json

```diff
@@ -105,6 +105,7 @@
     "tslint-react": "^2.4.0",
     "typescript": "^2.3.4",
     "typescript-babel-jest": "^1.0.3",
+    "typescript-styled-plugin": "^0.1.2",
```

Comment:
See https://github.com/Microsoft/typescript-styled-plugin. Provides intellisense and syntax-checking, but doesn't work on `StyledFunction` interface nodes (damn)

Reply:
Ah, nice - we'd been looking at adding this to emission, all our CSS recommendations are coming through from VS Code side rather than the language. I think this plugin is probably more reliable 👍

Reply:
I looked into the `StyledFunction` thing - it's because it's just looking at [every tagged string](https://github.com/Microsoft/typescript-styled-plugin/blob/master/src/index.ts#L28-L29) and seeing if the generator is [one of these](https://github.com/Microsoft/typescript-styled-plugin/blob/master/src/configuration.ts#L11)

Looks like it's not interpreting types itself, so the `x = StyledFunction(Y)` won't know x`` is a thing

Reply:
Ah yeah, thats what i figured :/